### ISSUE TITLE
Incorrect "Leftover tokens" error with `$(…)?` repetition

### ISSUE THREAD

**Issue by SludgePhD** [rust-lang/rust-analyzer#22431]
**rust-analyzer version**: 0.3.2845-standalone

**rustc version**: current stable, but shouldn't matter

**editor or extension**: VS Code

**relevant settings**: n/a

**code snippet to reproduce**:
```rust
#[macro_export]
macro_rules! attr {
    ($t:ty) => {
        {
            $crate::attr_array!($t; 1);
        }
    };
}

#[macro_export]
macro_rules! attr_array {
    ($( $t:ty; )? $count:expr) => {
        [(); $count]  // expected (), found [(); _]
    };
}

fn make<T>() {
    attr!(T)  // leftover tokens
}
```

rust-analyzer reports 2 incorrect errors, but rustc accepts this code fine.

Unlike the other issues that mention this particular error, my reproduction seems to depend on a `$(…)?` or `$(…)*` repetition being present (a `+` repetition makes the error go away), so I figured I'd file a separate issue.

**Comment by Abusalah0** [comment]
I can reproduce this with `rust-analyzer 0.3.2845-standalone`.

Environment:

```text
Linux abusalah0 7.0.10-201.fc44.x86_64 #1 SMP PREEMPT_DYNAMIC Wed May 27 13:57:41 UTC 2026 x86_64 GNU/Linux
rustc 1.96.0 (ac68faa20 2026-05-25) (Fedora 1.96.0-1.fc44)
cargo 1.96.0 (30a34c682 2026-05-25) (Fedora 1.96.0-1.fc44)
rust-analyzer 0.3.2845-standalone
```

The smallest compiler-accepted reproducer I found is:

```rust
macro_rules! attr {
    ($t:ty) => {{
        attr_array!($t; 1);
    }};
}

macro_rules! attr_array {
    ($( $t:ty; )? $count:expr) => {
        [(); $count]
    };
}

pub fn make<T>() {
    attr!(T)
}
```

`cargo check` accepts this:

```text
Checking ra-macro-mcve v0.1.0 (.../ra-macro-mcve)
Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.02s
```

But `rust-analyzer diagnostics .` reports these error diagnostics:

```text
Error Ra("macro-error", Error) from LineCol { line: 13, col: 4 } to LineCol { line: 13, col: 9 }: leftover tokens
Error SyntaxError from LineCol { line: 13, col: 4 } to LineCol { line: 13, col: 9 }: Syntax Error in Expansion: expected expression
Error SyntaxError from LineCol { line: 13, col: 4 } to LineCol { line: 13, col: 9 }: Syntax Error in Expansion: expected R_BRACK
Error SyntaxError from LineCol { line: 13, col: 4 } to LineCol { line: 13, col: 9 }: Syntax Error in Expansion: expected expression, item or let statement
Error RustcHardError("E0425") from LineCol { line: 8, col: 14 } to LineCol { line: 8, col: 19 }: no such value in this scope
Error RustcHardError("E0308") from LineCol { line: 8, col: 14 } to LineCol { line: 8, col: 19 }: expected (), found [(); _]
```

I also tested these variants:

| Variant | `cargo check` | `rust-analyzer diagnostics` |
|---|---:|---:|
| `$( $t:ty; )? $count:expr` through wrapper macro | passes | false diagnostics |
| `$( $t:ty; )* $count:expr` through wrapper macro | passes | false diagnostics |
| `$( $t:ty; )+ $count:expr` through wrapper macro | passes | no error diagnostics |
| `$t:ty; $count:expr` through wrapper macro | passes | no error diagnostics |

Some reduction notes:

- `#[macro_export]` and `$crate::` are not required for the reproduction.
- A direct call like `attr_array!(T; 1)` is not a smaller MCVE for me because `cargo check` rejects it as locally ambiguous.
- A generic type parameter is not required when the wrapper macro is kept; `attr!(u8)` still reproduces.
- Simplifying the expansion body to `$count` still produces false diagnostics, though the array-specific diagnostics disappear.

**Comment by ChayimFriedman2** [comment]
Another issue of transparent groups. Maybe we need a label for those...
